src/nni_manager/training_service/reusable/trialDispatcher.ts

```diff
@@ -462,7 +459,7 @@ class TrialDispatcher implements TrainingService {
             environment.workingFolder = storageService.joinPath("envs", envId);
             await storageService.createDirectory(environment.workingFolder);
         } else {
-            environment.command = `cd envs && sh install_nni.sh && mkdir ${envId} && cd ${envId} && python3 -m nni_trial_tool.trial_runner`;
+            environment.command = `mkdir envs/${envId} && cd envs/${envId} && sh ../install_nni.sh && python3 -m nni_trial_tool.trial_runner`;
```

Comment:
why not use environment.command like below?

environment.command = `mkdir envs/${envId} && cd envs/${envId} && ${environment.command}`;

Reply:
BTW, you can merge the whole logic with above also. Even for centralized storage service, it doesn't need to create this folder in TrialDispatcher. So after that the whole logic will be the same.

Reply:
fixed.